app/models/alchemy/page.rb

```diff
@@ -84,12 +101,29 @@ class Page < ActiveRecord::Base
     attr_accessor :do_not_sweep
     attr_accessor :do_not_validate_language
 
-    before_save :set_language_code, if: -> { language.present? }, unless: :systempage?
-    before_save :set_restrictions_to_child_pages, if: :restricted_changed?, unless: :systempage?
-    before_save :inherit_restricted_status, if: -> { parent && parent.restricted? }, unless: :systempage?
-    before_save :update_published_at, if: -> { public && read_attribute(:published_at).nil? }, unless: :systempage?
-    before_create :set_language_from_parent_or_default, if: -> { language_id.blank? }, unless: :systempage?
-    after_update :create_legacy_url, if: :urlname_changed?, unless: :redirects_to_external?
+    before_save :set_language_code,
+      if: -> { language.present? },
+      unless: :systempage?
+
+    before_save :set_restrictions_to_child_pages,
+      if: :restricted_changed?,
+      unless: :systempage?
+
+    before_save :inherit_restricted_status,
+      if: -> { parent && parent.restricted? },
+      unless: :systempage?
+
+    before_save :set_published_at,
+      if: -> { public? && published_at.nil? },
+      unless: :systempage?
+
+    before_create :set_language_from_parent_or_default,
+      if: -> { language_id.blank? },
+      unless: :systempage?
+
+    after_update :create_legacy_url,
+      if: :urlname_changed?,
+      unless: :redirects_to_external?
```

Comment:
It would've been nice to have these changes in a separate PR. Don't mess with it now, but for the future :)